### PR TITLE
AJAX pagination added and some updation in forms in list of project a…

### PR DESCRIPTION
**Pull request by haider-sheikh**
Ajax Pagination Added
Form wise errors handled
edit/show/delete buttons changed

### CODE REVIEW

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -40,19 +42,22 @@ def edit
   def update
     if @project.update(project_params)
       flash[:success] = t('flash_messages.update', name: t('shared.project'))
+      respond_to do |format|
+        format.js { redirect_to projects_path }
```
Comment (hafsabatool7) [inline app/controllers/projects_controller.rb]: Add option to edit and delete project from the show page as well

File: app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -40,19 +42,22 @@ def edit
...
+      respond_to do |format|
+        format.js
+      end
     end
   end
 
   def show
     add_breadcrumb @project.name, :project_path
-    @pagy, @project_issues = pagy(@project.issues, items: Company::PAGE_SIZE)
+    @pagy, @project_issues = pagy(@project.issues, link_extra: "data-remote='true'", items: Company::PAGE_SIZE)
```
Comment (hafsabatool7) [inline app/controllers/projects_controller.rb:57]: If this is to be used everywhere, move to pagy.rb

File: app/controllers/projects_controller.rb [hunk 3]
```diff
@@ -67,6 +72,13 @@ def destroy
     end
   end
 
+  def issues_list
```
Comment (hafsabatool7) [inline app/controllers/projects_controller.rb]: Why did we create a new action for issues_list? What is wrong with having a show.js.erb (Along with html.erb) to handle the js case?

File: app/controllers/projects_controller.rb [hunk 4]
```diff
@@ -75,7 +87,7 @@ def filters
         @projects = @projects.independent_projects
       end
     end
-    @pagy, @projects = pagy(@projects.order(created_at: :desc), items: Company::PAGE_SIZE)
+    load_pagy
```
Comment (hafsabatool7) [inline app/controllers/projects_controller.rb:83]: Why not add it to before_action too?

File: app/controllers/teams_controller.rb [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
+  before_action :load_pagy, only: %i[teams_table index]
+
   add_breadcrumb I18n.t('shared.home'), :root_path, only: [:index, :show]
   add_breadcrumb I18n.t('shared.teams'), :teams_path, only: [:index, :show]
 
   def index
-    @pagy, @teams = pagy(@teams.order(created_at: :desc), items: Company::PAGE_SIZE)
     respond_to do |format|
       format.html
+      format.js { render 'teams_table'}
```
Comment (hafsabatool7) [inline app/controllers/teams_controller.rb]: Same comment - why can't we handle this logic in index.js.erb?

File: app/controllers/teams_controller.rb [hunk 6]
```diff
@@ -39,12 +41,15 @@ def edit
   def update
     if @team.update(team_params)
       flash[:success] = t('flash_messages.update', name: t('shared.team'))
+      respond_to do |format|
```
Comment (hafsabatool7) [inline app/controllers/teams_controller.rb:44]: Add option to edit/delete team from its show page

File: app/views/layouts/_form_errors.html.erb [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<div class="project-errors">
```
Comment (hafsabatool7) [inline app/views/layouts/_form_errors.html.erb]:
```suggestion
<div class="form-errors">
```

File: app/views/projects/_index_table.html.erb [hunk 8]
```diff
@@ -20,15 +20,15 @@
           <td><%= project.project_category%></td>
           <td><%= time_ago_in_words(project.created_at) %> <%= t('shared.ago') %></td>
           <td>
+            <%= link_to '<i class="fas fa-eye"></i>'.html_safe,project_path(project), class: 'btn btn-outline-info btn-sm card-button' %>
             <% if can? [:destroy, :update], project %>
-              <%= link_to t('shared.delete'), project_path(project), method: :delete, class: 'btn btn-danger' %>
-              <%= link_to t('shared.edit'), edit_project_path(project), {class: 'btn my-primary-btn',remote: true, 'data-toggle' =>  "modal", 'data-target' => '#create_project_modal'} %>
+              <%= link_to '<i class="fas fa-pencil-alt"></i>'.html_safe, edit_project_path(project), {class: 'btn btn-outline-info btn-sm card-button mx-2',remote: true, 'data-toggle' =>  "modal", 'data-target' => '#create_project_modal'} %>
+              <%= link_to '<i class="fas fa-trash-alt"></i>'.html_safe, project_path(project), method: :delete, class: 'btn btn-sm btn-outline-danger card-button' %>
```
Comment (hafsabatool7) [inline app/views/projects/_index_table.html.erb]: Use link_to with a block

File: config/locales/en.yml [hunk 9]
```diff
@@ -216,6 +216,8 @@ en:
     project_categories_team: 'Team'
     project_categories_independent: 'Independent'
   projects:
+    input_name: 'Name*'
+    input_category: 'Category*'
```
Comment (hafsabatool7) [inline config/locales/en.yml:233]: Not actually a good practice to write asterisk used for required in internationalized text.

File: app/controllers/projects_controller.rb [hunk 10]
```diff
@@ -40,19 +42,22 @@ def edit
...
-    end
-    respond_to do |format|
-      format.js { redirect_to projects_path }
+      respond_to do |format|
+        format.js
+      end
     end
   end
 
   def show
```
Comment (hafsabatool7) [inline app/controllers/projects_controller.rb:55]: To display the list of teams, use the same partial of teams table this is used on teams index page. Similarly for issues and users.

File: app/controllers/projects_controller.rb (hunk 4 above)
Comment (haider-sheikh) [reply app/controllers/projects_controller.rb:83]: Becoz @project k andr data change ho raha ha

File: app/controllers/projects_controller.rb (hunk 2 above)
Comment (haider-sheikh) [reply app/controllers/projects_controller.rb:57]: will do in next PR

File: config/locales/en.yml (hunk 9 above)
Comment (haider-sheikh) [reply config/locales/en.yml:233]: will do in next PR

File: app/controllers/projects_controller.rb (hunk 10 above)
Comment (haider-sheikh) [reply app/controllers/projects_controller.rb:55]: will do in next PR

**Pull request merged**